src/x/query.rs

```diff
@@ -98,3 +133,44 @@ where
         }
     }
 }
+
+/// A meta [Query] for combining two queries with a logical AND.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct AndQuery<X: XConn, Q1, Q2> {
+    first: Q1,
+    second: Q2,
+    _phantom: std::marker::PhantomData<X>,
+}
+
+impl<X: XConn, Q1: Query<X>, Q2: Query<X>> Query<X> for AndQuery<X, Q1, Q2> {
```

Comment:
If we are sticking with the trait bounds here instead of using trait objects can we use `where` please? Once there is more than a single bound the inline bounds get pretty tricky to parse visually.

```rust
impl<X, Q1, Q2> Query<X> for AndQuery<X, Q1, Q2>
where
  X: XConn,
  Q1: Query<X>,
  Q2: Query<X>,
{
    ...
}
```

Reply:
Yeah, I'm happy with internal dynamic dispatch. So I'm not going to worry about this.